tests/pocket_test.cpp

```diff
@@ -0,0 +1,218 @@
+
+    pocket_data data_corpse( item_pocket::pocket_type::CORPSE );
+
+    GIVEN( "a corpse" ) {
+        item_pocket pocket_corpse( &data_corpse );
+
+        THEN( "it can contain items" ) {
+            item rock( "test_rock" );
+            item glock( "test_glock" );
+
+            expect_can_contain( pocket_corpse, rock );
+            expect_can_contain( pocket_corpse, glock );
+        }
+    }
+}
+
+TEST_CASE( "holster can contain one fitting item", "[pocket][holster]" )
+{
+    // Start with a basic test handgun from data/mods/TEST_DATA/items.json
+    item glock( "test_glock" );
+
+    // Construct data for a holster to perfectly fit this gun
+    pocket_data data_holster( item_pocket::pocket_type::CONTAINER );
+    data_holster.holster = true;
+    data_holster.volume_capacity = glock.volume();
+    data_holster.max_item_length = glock.length();
+    data_holster.max_contains_weight = glock.weight();
+
+    GIVEN( "holster has enough volume, length, and weight capacity" ) {
+        // Use the perfectly-fitted holster data
+        item_pocket pocket_holster( &data_holster );
+
+        THEN( "it can contain the gun" ) {
+            expect_can_contain( pocket_holster, glock );
+        }
+    }
+
+    GIVEN( "holster has not enough weight capacity" ) {
+        // Reduce holster weight capacity by 1 gram
+        data_holster.max_contains_weight = glock.weight() - 1_gram;
+        item_pocket pocket_holster( &data_holster );
+
+        THEN( "it cannot contain the gun, because it is too heavy" ) {
+            expect_cannot_contain( pocket_holster, glock, "item is too heavy",
+                                   item_pocket::contain_code::ERR_TOO_HEAVY );
+        }
+
+    }
+
+    GIVEN( "holster has not enough volume capacity" ) {
+        // Reduce holster volume capacity by 1 ml
+        data_holster.volume_capacity = glock.volume() - 1_ml;
+        item_pocket pocket_holster( &data_holster );
+
+        THEN( "it cannot contain the gun, because it is too big" ) {
+            expect_cannot_contain( pocket_holster, glock, "item too big",
+                                   item_pocket::contain_code::ERR_TOO_BIG );
+        }
+    }
+
+    GIVEN( "holster already contains an item" ) {
+        item other_glock( "test_glock" );
+        item_pocket pocket_holster( &data_holster );
+        pocket_holster.insert_item( other_glock );
+
+        THEN( "it cannot contain the gun, because holster can only hold one item" ) {
+            expect_cannot_contain( pocket_holster, glock, "holster already contains an item",
+                                   item_pocket::contain_code::ERR_NO_SPACE );
+        }
+    }
+}
+
+TEST_CASE( "magazine with ammo restriction", "[pocket][magazine][ammo_restriction]" )
+{
+    // Generic magazine with plenty of capacity for almost any ammo
+    pocket_data data_mag( item_pocket::pocket_type::MAGAZINE );
+    data_mag.volume_capacity = 10_liter;
```

Comment:
i recommend you actually test with these set at default values/ 0, since that's how they mostly get loaded into the json (itemfactory is just being special for now)

Reply:
Thanks, and confirmed - even explicit 0 capacity works, so I've included that in the last revision with an explanaion.